tracing-subscriber/src/registry/sharded.rs

```diff
@@ -541,10 +541,6 @@ mod tests {
         Collect,
     };
 
-    #[derive(Debug)]
-    struct DoesNothing;
-    impl<C: Collect> Subscribe<C> for DoesNothing {}
```

Comment:
I can't recall what this code was intended to test...I wonder if it was testing that a trivial `Subscribe` impl compiles? But because we have other implementations in the test suite, I think deleting it is fine.